lib/miq_expression/tag.rb

```diff
@@ -1,6 +1,6 @@
 class MiqExpression::Tag < MiqExpression::Target
   REGEX = /
-(?<model_name>([[:alnum:]]*(::)?)?)
+(?<model_name>([[:alnum:]]*(::)?)*)
```

Comment:
@yrudman you need probably limit count of `::`
Problem is failing case like
```ruby
pression::Tag.parse('ManageIQ::Providers::CloudManagermanaged-se')
```

Reply:
👍   nice catch @lpichler 
parsing your example from console never finished and was interrupted after 2 minutes. 
And parsing it on master returning `nil`.  So, limiting `::` looks like good idea.